packages/x-adapter/src/__tests__/endpoint-adapter.factory.spec.ts

```diff
@@ -0,0 +1,268 @@
+import { endpointAdapterFactory } from '../endpoint-adapter.factory';
+import { identityMapper } from '../mappers/identity.mapper';
+import { EndpointAdapter, EndpointAdapterOptions } from '../types/adapter.types';
+import { HttpClient, RequestOptions } from '../types/http-client.types';
+import { Mapper } from '../types/mapper.types';
+
+/**
+ * Creates an {@link EndpointAdapter} using the {@link endpointAdapterFactory} with the given
+ * {@link EndpointAdapterOptions}, exposing a basic API for testing.
+ *
+ * @param options - The CreateEndpointAdapterFactoryOptions to create the API with.
+ *
+ * @returns The API for testing the {@link EndpointAdapter}.
+ */
+function createEndpointAdapterFactoryOptions<Request, Response>({
+  options,
+  rawResponse = {} as Response
+}: CreateEndpointAdapterFactoryOptions<Request, Response> = {}): CreateEndpointAdapterFactoryAPI<
+  Request,
+  Response
+> {
+  const mockedHttpClient = jest.fn(() => Promise.resolve(rawResponse));
+  const mockedRequestMapper = jest.fn(identityMapper);
+  const mockedResponseMapper = jest.fn(identityMapper);
+
+  const adapterOptions: EndpointAdapterOptions<Request, Response> = {
+    endpoint: 'https://api.empathy.co/test',
+    httpClient: mockedHttpClient as HttpClient,
+    requestMapper: mockedRequestMapper,
+    responseMapper: mockedResponseMapper,
+    ...options
+  };
+  const endpointAdapter = endpointAdapterFactory<Request, Response>(adapterOptions);
+
+  return {
+    endpointAdapter,
+    options: adapterOptions,
+    mockedHttpClient,
+    mockedRequestMapper,
+    mockedResponseMapper
+  };
+}
+
+describe('adapterFactory tests', () => {
+  const request: TestRequest = { q: 'patata' };
+
+  it('should create an endpointAdapter with the given options', async () => {
+    const rawResponse: TestResponse = { query: 'patata' };
+    const {
+      endpointAdapter,
+      options,
+      mockedHttpClient,
+      mockedRequestMapper,
+      mockedResponseMapper
+    } = createEndpointAdapterFactoryOptions<TestRequest, TestResponse>({
+      rawResponse
+    });
+
+    expect(endpointAdapter).toBeDefined();
+    expect(endpointAdapter).toHaveProperty('extends');
+
+    const response = await endpointAdapter(request);
+
+    expect(mockedRequestMapper).toHaveBeenCalledTimes(1);
+    expect(mockedHttpClient).toHaveBeenCalledTimes(1);
+    expect(mockedHttpClient).toHaveBeenCalledWith(options.endpoint, {
+      parameters: request
+    });
+    expect(mockedResponseMapper).toHaveBeenCalledTimes(1);
+    expect(response).toEqual(rawResponse);
+  });
+
+  it('should extend the initial endpointAdapter options with the new ones', async () => {
```

Comment:
Maybe it is too much but I think would be nice to add some check that extending the adapter does not modify the original one.